Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -314,23 +324,46 @@ void OpenALStream::SoundLoop()
     // many silence samples.  These do not need to be timestretched.
     if (rate > 0.10)
     {
-      soundTouch.setTempo(rate);
-      if (rate > 10)
+      if (use_timestretching)
+      {
+        soundTouch.setTempo(rate);
+      }
+      else
+      {
+        alSourcef(ui_source, AL_PITCH, static_cast<ALfloat>(rate));
```

Comment:
Sorry, maybe I am not understanding which is which.
I understand pitch-shifting as changing the tempo without correcting the pitch, when the emulation speed is slow the tempo is reduced and the pitch gets lower (gets shifted). This operation is fast.
Time-stretching I understand as changing the tempo and pitch-correcting the output, which is slow but outputs audio at the correct pitch.
If my understanding is wrong, what are the correct terms?